File: process-images.js
/**
 * Script para procesar imágenes: remover fondo y reescalar
 * Usa @imgly/background-removal (funciona en Node.js)
 * 
 * Instalación:
 * npm install @imgly/background-removal sharp
 * 
 * Uso:
 * node process-images.js
 */

const fs = require('fs').promises;
const path = require('path');
const sharp = require('sharp');
const { removeBackground } = require('@imgly/background-removal-node');

const INPUT_DIR = path.join(__dirname, 'client', 'public', 'imagenes_hoco_productos');
const OUTPUT_DIR = path.join(__dirname, 'client', 'public', 'imagenes_hoco_productos_sin_fondo');
const TARGET_SIZE = 1200;

async function processImage(inputPath, outputPath) {
  try {
    const filename = path.basename(inputPath);
    console.log(`Procesando: ${filename}`);
    
    // Leer imagen
    const imageBuffer = await fs.readFile(inputPath);
    
    // Remover fondo
    const blob = await removeBackground(imageBuffer);
    const arrayBuffer = await blob.arrayBuffer();
    const buffer = Buffer.from(arrayBuffer);
    
    // Procesar con sharp: redimensionar y centrar
    const image = sharp(buffer);
    const metadata = await image.metadata();
    
    // Calcular dimensiones manteniendo aspect ratio
    const scale = Math.min(TARGET_SIZE / metadata.width, TARGET_SIZE / metadata.height);
    const newWidth = Math.round(metadata.width * scale);
    const newHeight = Math.round(metadata.height * scale);
    
    // Redimensionar y agregar padding para centrar
    await image
      .resize(newWidth, newHeight, {
        fit: 'contain',
        background: { r: 0, g: 0, b: 0, alpha: 0 }
      })
      .extend({
        top: Math.floor((TARGET_SIZE - newHeight) / 2),
        bottom: Math.ceil((TARGET_SIZE - newHeight) / 2),
        left: Math.floor((TARGET_SIZE - newWidth) / 2),
        right: Math.ceil((TARGET_SIZE - newWidth) / 2),
        background: { r: 0, g: 0, b: 0, alpha: 0 }
      })
      .png({ quality: 95, compressionLevel: 9 })
      .toFile(outputPath);
    
    console.log(`✓ Guardado: ${path.basename(outputPath)}`);
    return true;
  } catch (error) {
    console.error(`✗ Error procesando ${path.basename(inputPath)}: ${error.message}`);
    return false;
  }
}

async function main() {
  try {
    // Crear directorio de salida
    await fs.mkdir(OUTPUT_DIR, { recursive: true });
    
    // Obtener todas las imágenes
    const files = await fs.readdir(INPUT_DIR);
    const imageFiles = files.filter(file => 
      /\.(png|jpg|jpeg)$/i.test(file)
    );
    
    console.log(`Encontradas ${imageFiles.length} imágenes para procesar`);
    console.log(`Directorio de salida: ${OUTPUT_DIR}`);
    console.log('-'.repeat(50));
    
    let successCount = 0;
    
    // Procesar cada imagen
    for (let i = 0; i < imageFiles.length; i++) {
      const inputPath = path.join(INPUT_DIR, imageFiles[i]);
      const outputPath = path.join(OUTPUT_DIR, imageFiles[i].replace(/\.(jpg|jpeg)$/i, '.png'));
      
      console.log(`[${i + 1}/${imageFiles.length}] `);
      
      if (await processImage(inputPath, outputPath)) {
        successCount++;
      }
    }
    
    console.log('-'.repeat(50));
    console.log(`✓ Completado: ${successCount}/${imageFiles.length} imágenes procesadas`);
    console.log(`Imágenes guardadas en: ${OUTPUT_DIR}`);
    console.log('\nPara usar las nuevas imágenes:');
    console.log('1. Revisa las imágenes procesadas');
    console.log('2. Reemplaza las originales o actualiza las rutas');
    
  } catch (error) {
    console.error('Error:', error);
  }
}

main();

File: docs/process_images.py
"""
Script para procesar imágenes de productos:
1. Remover fondo (convertir a PNG transparente)
2. Reescalar para mejor resolución

Requisitos:
pip install rembg pillow

Uso:
python process_images.py
"""

import os
from pathlib import Path
from PIL import Image
from rembg import remove

# Configuración
INPUT_DIR = Path("client/public/imagenes_hoco_productos")
OUTPUT_DIR = Path("client/public/imagenes_hoco_productos_processed")
TARGET_SIZE = (1200, 1200)  # Tamaño objetivo (ancho, alto)
QUALITY = 95  # Calidad de compresión (1-100)

def process_image(input_path, output_path):
    """Procesa una imagen: remueve fondo y reescala"""
    try:
        print(f"Procesando: {input_path.name}")
        
        # Abrir imagen
        with open(input_path, 'rb') as input_file:
            input_data = input_file.read()
        
        # Remover fondo
        output_data = remove(input_data)
        
        # Abrir imagen sin fondo
        img = Image.open(io.BytesIO(output_data))
        
        # Convertir a RGBA si no lo está
        if img.mode != 'RGBA':
            img = img.convert('RGBA')
        
        # Calcular nuevo tamaño manteniendo aspect ratio
        img.thumbnail(TARGET_SIZE, Image.Resampling.LANCZOS)
        
        # Crear imagen con fondo transparente del tamaño objetivo
        new_img = Image.new('RGBA', TARGET_SIZE, (0, 0, 0, 0))
        
        # Centrar imagen
        x = (TARGET_SIZE[0] - img.width) // 2
        y = (TARGET_SIZE[1] - img.height) // 2
        new_img.paste(img, (x, y), img)
        
        # Guardar
        new_img.save(output_path, 'PNG', optimize=True, quality=QUALITY)
        print(f"✓ Guardado: {output_path.name}")
        
    except Exception as e:
        print(f"✗ Error procesando {input_path.name}: {str(e)}")

def main():
    # Crear directorio de salida
    OUTPUT_DIR.mkdir(exist_ok=True)
    
    # Obtener todas las imágenes PNG
    image_files = list(INPUT_DIR.glob("*.png")) + list(INPUT_DIR.glob("*.jpeg")) + list(INPUT_DIR.glob("*.jpg"))
    
    print(f"Encontradas {len(image_files)} imágenes para procesar")
    print(f"Directorio de salida: {OUTPUT_DIR}")
    print("-" * 50)
    
    # Procesar cada imagen
    for i, input_path in enumerate(image_files, 1):
        output_path = OUTPUT_DIR / f"{input_path.stem}.png"
        print(f"[{i}/{len(image_files)}] ", end="")
        process_image(input_path, output_path)
    
    print("-" * 50)
    print(f"✓ Procesamiento completado!")
    print(f"Imágenes guardadas en: {OUTPUT_DIR}")
    print("\nPara usar las nuevas imágenes:")
    print("1. Revisa las imágenes procesadas")
    print("2. Reemplaza las originales o actualiza las rutas en el código")

if __name__ == "__main__":
    import io
    main()
